### PR TITLE
[FLINK-34020] Bump CI flink version on flink-connector-rabbitmq

### PR DESCRIPTION
**Pull request by MartijnVisser**

### CODE REVIEW

**Comment by MartijnVisser** [comment]
@snuyanzin Any ideas why everything passes for the non 1.19-SNAPSHOT tests, but nothing for 1.19-SNAPSHOT?

**Comment by snuyanzin** [comment]
The problem is with 
`org.apache.flink.streaming.connectors.rabbitmq.RMQSourceTest`
especially with `org.apache.flink.streaming.connectors.rabbitmq.RMQSourceTest#testDuplicateId` which became endless...

if you comment it it will stop hanging however there are some tests which will fail...
Need to investigate what changes in 1.19 or in deps (e.g. result of converge deps fixes) leads to such behavior

**Comment by MartijnVisser** [comment]
> if you comment it it will stop hanging however there are some tests which will fail...

The weird thing is that it passes flawless locally. I can't get this reproduced locally

**Comment by snuyanzin** [comment]
I was able to reproduce it locally with 
```
mvn clean install -Dflink.version=1.19-SNAPSHOT
```

**Comment by snuyanzin** [comment]
local test shows that this is the first commit 
```
commit e9353319ad625baa5b2c20fa709ab5b23f83c0f4 (HEAD)
Author: Wencong Liu <liuwenclever@163.com>
Date:   Fri Sep 8 11:04:56 2023 +0800

    [FLINK-32978][flink-core] Migrate the usage of RichFunction#open(Configuration parameters) to RichFunction#open(OpenContext openContext)
    
    This closes #23058

```
when it stopped workiing for flink connector rabbitmq
probably it is the reason

**Comment by snuyanzin** [comment]
I commented about that at FLINK-32978 

also I submitted a PR https://github.com/MartijnVisser/flink-connector-rabbitmq/pull/1 to your branch applying WA to work after changes within FLINK-32978   and making it passing tests 
https://github.com/snuyanzin/flink-connector-rabbitmq/actions/runs/7453993758

File: flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSource.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 import org.apache.flink.api.common.serialization.DeserializationSchema;
 import org.apache.flink.api.common.serialization.RuntimeContextInitializationContextAdapters;
 import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.operators.translation.RichCombineToGroupCombineWrapper;
```
Comment (snuyanzin) [inline flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSource.java]:
```suggestion
```

File: flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSource.java [hunk 2]
```diff
@@ -236,7 +239,10 @@ protected void setupQueue() throws IOException {
     @Override
     public void open(Configuration config) throws Exception {
         super.open(config);
+        sessionIds = new ArrayList<>(64);
+        sessionIdsPerSnapshot = new ArrayDeque<>();
         try {
+            RichCombineToGroupCombineWrapper s;
```
Comment (snuyanzin) [inline flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSource.java]:
```suggestion
```

**Review by snuyanzin — APPROVED** [review]
Oops these 2 lines are useless, we could remove them

LGTM

**Pull request merged**